### PR TITLE
regression/compatibility testing

### PR DESCRIPTION
**Pull request by willscott**
* [x] add a binary that evaluates compatibility of two different drand binaries
  * [x] can a mix of the two versions perform DKG?
  * [x] can a new binary be reshared into an existing group?
  * [x] can a member of a current group upgrade to the new binary seamlessly?
* [x] Hook binary into CI

### CODE REVIEW

File: .github/workflows/regression.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      if: steps.cache.outputs.cache-hit != 'true'
+      run: go get ./...
+    - name: Dir Setup
+      run: mkdir -p ~/go/bin
+    - name: Build
+      run: go build . && cp drand ~/go/bin/drand-candidate
+
+    - uses: actions/checkout@v2
+      with:
+        ref: ''
```
[the remark this answers is not in the document]
Comment (willscott) [reply .github/workflows/regression.yml:30]: i hear the default branch from github may eventually change its name?

[the remark this answers is not in the document]
Comment (alanshaw) [reply .github/workflows/regression.yml:30]: The docs say the default for this value is the branch from where the PR came from. So i figured that unless we specify master or main or whatever our default branch is then we’re not testing against it. Am I missing something? Does empty mean default value (the PR’s branch) or default branch?

[the remark this answers is not in the document]
Comment (willscott) [reply .github/workflows/regression.yml]: good catch. changed to be explicit

File: cmd/relay-gossip/README.md [hunk 2]
```diff
@@ -36,7 +57,7 @@ Publish topic=/drand/pubsub/v0.0.0/<chain-hash> data={randomness}
    +-------------------------+   +-------------------------+
 ```
 
-<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- START doctoc doctoc doctoc generated TOC please keep comment here to allow auto update -->
```
Comment (alanshaw) [inline cmd/relay-gossip/README.md:60]: doctoc doctoc doctoc doctoc doctoc \m/

**Pull request merged**